pkg/kubelet/kubeletconfig/checkpoint/download.go

```diff
@@ -60,26 +87,10 @@ type RemoteConfigSource interface {
 	object() interface{}
 }
 
-// NewRemoteConfigSource constructs a RemoteConfigSource from a v1/NodeConfigSource object, or returns
-// a sanitized failure reason and an error if the `source` is blatantly invalid.
-// You should only call this with a non-nil config source.
-func NewRemoteConfigSource(source *apiv1.NodeConfigSource) (RemoteConfigSource, string, error) {
-	// exactly one subfield of the config source must be non-nil, toady ConfigMapRef is the only reference
-	if source.ConfigMapRef == nil {
-		return nil, status.FailSyncReasonAllNilSubfields, fmt.Errorf("%s, NodeConfigSource was: %#v", status.FailSyncReasonAllNilSubfields, source)
-	}
-
-	// validate the NodeConfigSource:
-
-	// at this point we know we're using the ConfigMapRef subfield
-	ref := source.ConfigMapRef
-
-	// name, namespace, and UID must all be non-empty for ConfigMapRef
-	if ref.Name == "" || ref.Namespace == "" || string(ref.UID) == "" {
-		return nil, status.FailSyncReasonPartialObjectReference, fmt.Errorf("%s, ObjectReference was: %#v", status.FailSyncReasonPartialObjectReference, ref)
-	}
-
-	return &remoteConfigMap{source}, "", nil
+// NewRemoteConfigSource constructs a RemoteConfigSource from a v1/NodeConfigSource object.
+// We assume the source has been validated beforehand - e.g. by an API server.
+func NewRemoteConfigSource(source *apiv1.NodeConfigSource) RemoteConfigSource {
+	return &remoteConfigMap{source}
```

Comment:
actually still need to check not all known fields nil, wrt old clients against a new API server that allows a new config source type failing cleanly (i.e. not crashing due to nil pointer dereference).